invenio_records_rest/serializers/response.py

```diff
@@ -32,7 +29,6 @@ def view(pid, record, code=200, headers=None, links_factory=None):
             mimetype=mimetype,
         )
         response.status_code = code
-        set_headers_for_record_caching_and_concurrency(response, record)
```

Comment:
I'm not sure this is correct, we still need this data to pass to the `PUT` for concurrency control. So I think removing only the check from the `GET` method is the way to go (browser will still send the header, but we'll never respond with a 304).

Reply:
You might be right... :+1: